core/db/entities.py

```diff
@@ -1,58 +1,65 @@
 import uuid  # noqa
+from typing import List
 
 import sqlalchemy as sqla
+from sqlalchemy.orm import Mapped, class_mapper
 
 from core import const
 from core.db import db
 from core.db.types import GUID
 
 
-class Organisation(db.Model):
+class BaseModel(db.Model):
+    __abstract__ = True
+
+    id: Mapped[int] = sqla.orm.mapped_column(
```

Comment:
I know this is a "pythonic" way of reducing the repeatability of defining the same id definition in every table, but I definitely think there's an argument for keeping model definitions verbose and individually defined. 

In this case I prefer the verbose way, as they are easy to read at a glance, and also easy to change if we, for example, decide to switch some PK's to be natural keys rather than surrogates. 

But happy to discuss. @cyrusdobbs @Joel-Pearce @gidsg @bobbi-hbrown any preference?

Reply:
Hmm yeah I get you. I do think the state of the models is still quite clear because _every_ model inherits `BaseModel` and has an `id` field. I think if some models did inherit `id` and some didn't then I think the argument for verboseness is stronger.

If we switched to natural keys then it would definitely have to change.

I'm happy for it to be kept like this or change it - I don't really have strong feelings.

Reply:
will resolve for now, we can revisit if it becomes a problem